source/MaterialXCore/Definition.cpp

```diff
@@ -186,7 +186,7 @@ StringVec TargetDef::getMatchingTargets() const
 vector<UnitDefPtr> UnitTypeDef::getUnitDefs() const
 {
     vector<UnitDefPtr> unitDefs;
-    for (UnitDefPtr unitDef : getDocument()->getChildrenOfType<UnitDef>())
+    for (UnitDefPtr unitDef : getDocument()->getChildrenOfType<UnitDef>(EMPTY_STRING, getDocument()->getDataLibrary()))
```

Comment:
@ashwinbhat This latest work looks really close to the mark, and my one remaining question is whether we should always include the data library when calling getChildOfType or getChildrenOfType on a Document. Although there are a few edge cases where we omit the data library for Document calls, they actually look like bugs that we'll need to fix in the future, rather than intentional choices in the current proposal.

If my sense is correct, and we actually want to provide the data library universally when calling these methods on a Document, then we can greatly simplify this proposed change in the following way:

Remove the dataLibrary argument from both getChildOfType and getChildrenOfType.
Within each of these methods, check whether the current element is a Document. If we find that it's a Document, then the data library referenced by the document is included in the search; otherwise, no data library is included.
I think this would cover all cases that the current proposal does, but in a more universal, simplified, and bulletproof way. Let me know your thoughts!